### ISSUE TITLE
Peer dependencies not exhaustive

### ISSUE THREAD

**Issue by JamesYeoman** [storybookjs/builder-vite#235]
Using the berry version of yarn, with the improved install logging that is default on the berry version, sb-builder-vite is being flagged as not providing react (peer dependency for @storybook/source-loader), react-dom (peer dependency for @storybook/source-loader), and typescript (peer dependency for @joshwooding/vite-plugin-react-docgen-typescript).

![image](https://user-images.githubusercontent.com/21042602/153581165-76e57f09-7cb0-4398-b375-d1ac2648920d.png)

Yarn even makes the error codes clickable to open the [yarn docs page for the error code](https://yarnpkg.com/advanced/error-codes#yn0002---missing_peer_dependency).

TL;DR yarn enforces peer deps at every level of the dep tree, which results in peer dependency hoisting problems (as described by [this blog post](https://dev.to/arcanis/implicit-transitive-peer-dependencies-ed0) the yarn page linked

**Comment by domq** [comment]
FWIW, switching to Yarn “berry” version makes this symptom go away mostly. Perhaps this requirement of Yarn “classic” is now obsolete, and therefore this issue should be closed?
